### PR TITLE
[#1385] feat(api): Add Decimal class for holding decimal literal value

### PR DESCRIPTION
**Pull request by mchades**
### What changes were proposed in this pull request?

Add Decimal class for holding decimal literal value

### Why are the changes needed?

Fix: #1385 

### Does this PR introduce _any_ user-facing change?

yes, users now can use decimal literal conveniently

### How was this patch tested?

UTs added

### CODE REVIEW

File: api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  /** @return The scale of the decimal. */
+  public int scale() {
+    return scale;
+  }
+
+  private final BigDecimal value;
+  private final int precision;
+  private final int scale;
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java:78]: Can you please move the class fields to the top of the class.

File: api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+import static com.datastrato.gravitino.rel.types.Types.DecimalType.checkPrecisionScale;
+
+import com.google.common.base.Preconditions;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.Objects;
+
+/** Used to represent a {@link Types.DecimalType} value in Gravitino. */
+public class Decimal {
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java:15]: Why do we need a `Decimal` value in Gravitino, can you please add more comments?

File: api/src/main/java/com/datastrato/gravitino/rel/expressions/literals/Literals.java [hunk 3]
```diff
@@ -98,6 +99,17 @@ public static LiteralImpl<Double> doubleLiteral(Double value) {
     return of(value, Types.DoubleType.get());
   }
 
+  /**
+   * Creates a decimal type literal with the given value.
+   *
+   * @param value the decimal literal value
+   * @return a new {@link Literal} instance
+   */
+  public static LiteralImpl<Decimal> decimal(Decimal value) {
```
Comment (jerryshao) [inline api/src/main/java/com/datastrato/gravitino/rel/expressions/literals/Literals.java:108]: Is it better renaming to `decimalLiteral`?

File: api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java (hunk 2 above)
Comment (mchades) [reply api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java:15]: At first, I intended to use the type name directly as the method name, such as `long()`, `float()`, `double()`. However, as they are reserved keywords in Java, I had to resort to using `longLiteral()`, `floatLiteral()`, `doubleLiteral()`. 

If we want to use `decimalLiteral()` here, should the already defined methods like `time()`, `date()`, `string()` be changed to `xxxLiteral()`?

File: api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java (hunk 1 above)
Comment (mchades) [reply api/src/main/java/com/datastrato/gravitino/rel/types/Decimal.java]: done

File: api/src/main/java/com/datastrato/gravitino/rel/expressions/literals/Literals.java (hunk 3 above)
Comment (mchades) [reply api/src/main/java/com/datastrato/gravitino/rel/expressions/literals/Literals.java:108]: At first, I intended to use the type name directly as the method name, such as `long()`, `float()`, `double()`. However, as they are reserved keywords in Java, I had to resort to using `longLiteral()`, `floatLiteral()`, `doubleLiteral()`. 

If we want to use `decimalLiteral()` here, should the already defined methods like `time()`, `date()`, `string()` be changed to `xxxLiteral()`?

Comment (mchades) [reply api/src/main/java/com/datastrato/gravitino/rel/expressions/literals/Literals.java]: Already rename to `xxxLiteral()` for maintaining a consistent naming style

**Pull request merged**